### PR TITLE
Add ability to disable executions via a Callback

### PR DESCRIPTION
**Pull request by Neoministein**
The Idea of this pull request is to add the functionality to disable the act of executing somebody.

How to use the functionality:
```java

//You can add a callbackmethod which is called after pressing the melee button
AddCallback_IsValidMeleeExecutionTarget( DisableExectionAsPilot )

//For example: disables executions if the player is a pilot
bool function DisableExectionAsPilot(entity attacker, entity target)
{
  if (attacker.IsTitan()) {
    return true
  } else {
    return false
  }
}
```

My changes:

 - Added the `sh_melee.gnut` file from the base TF2 game.
 - Added a callback `AddCallback_IsValidMeleeExecutionTarget` which takes in a `bool functionref( entity attacker, entity target )`
 - At the bottom of `CodeCallback_IsValidMeleeExecutionTarget( entity attacker, entity target )` I loop through all the callbacks and if one returns false I block the execution from happening

### CODE REVIEW

File: Northstar.Custom/mod/scripts/vscripts/melee/sh_melee.gnut [hunk 1]
```diff
@@ -0,0 +1,1225 @@
...
+	SyncedMelee ornull action = FindBestSyncedMelee( attacker, target, actions )
+	if ( action == null )
+		return false
+
+	if ( !PlayerMelee_IsExecutionReachable( attacker, target, 0.3 ) )
+		return false
+
+	foreach ( callbackFunc in file.isValidMeleeExecutionTargetCallBacks )
+	{
+		if(!callbackFunc( attacker, target )) {
```
Comment (uniboi) [inline Northstar.Custom/mod/scripts/vscripts/melee/sh_melee.gnut:321]: formatting :)
```suggestion
		if( !callbackFunc( attacker, target ) )
		{
```

**Review by uniboi — APPROVED** [review]
Code looks good & tested with `script AddCallback_IsValidMeleeExecutionTarget(bool function(entity a, entity b){return false});ent_create npc_soldier; ent_fire !picker setteam 2`. Execution prompt still pops up but nobody is able to execute anything

**Comment by ASpoonPlaysGames** [comment]
> The callbacks get triggered after the melee button is pressed. I couldn't find the place in the code where the check occurs for the popup.

This is handled in native and from looking at the function that handles it, absolutely not worth hooking.

File: Northstar.Custom/mod/scripts/vscripts/melee/sh_melee.gnut [hunk 2]
```diff
@@ -310,6 +316,14 @@ bool function CodeCallback_IsValidMeleeExecutionTarget( entity attacker, entity
 	if ( !PlayerMelee_IsExecutionReachable( attacker, target, 0.3 ) )
 		return false
 
+	foreach ( callbackFunc in file.isValidMeleeExecutionTargetCallBacks )
+	{
+		if( !callbackFunc( attacker, target ) )
```
Comment (ASpoonPlaysGames) [inline Northstar.Custom/mod/scripts/vscripts/melee/sh_melee.gnut:321]:
```suggestion
		if ( !callbackFunc( attacker, target ) )
```
Just a formatting nitpick

**Review by ASpoonPlaysGames — APPROVED** [review]
Got a small formatting issue, but tbh I don't care about it enough to stop approval

**Comment by Neoministein** [comment]
Since this has already got 2 approvals, can this be merged?

**Comment by GeckoEidechse** [comment]
Thanks for the poke, just looked at it and there's a lot of code added. I assume this adds a new vanilla file?

Cause then I should add that one first ^^

**Comment by ASpoonPlaysGames** [comment]
It does, first commit of the PR

**Comment by GeckoEidechse** [comment]
Aight, will add that one separately rq ^^

**Comment by GeckoEidechse** [comment]
Aight, added vanilla file and update PR branch @ASpoonPlaysGames or @uniboi or @Neoministein or someone else double check it still looks alright and then we merge :D

**Comment by Neoministein** [comment]
Double checkt it, LGTM

**Pull request merged**